src/catch2/internal/catch_test_macro_impl.hpp

```diff
@@ -47,7 +47,7 @@
 
 ///////////////////////////////////////////////////////////////////////////////
 #define INTERNAL_CATCH_TEST( macroName, resultDisposition, ... ) \
-    do { /* NOLINT(bugprone-infinite-loop) */ \
+    do { /* NOLINT(bugprone-infinite-loop, cppcoreguidelines-avoid-do-while) */ \
```

Comment:
I'll merge this, but I am honestly getting tired of clang-tidy complaining about perfectly fine code.

I am kinda-ok with it being unable to reason about the loop enough to not see that it is not infinite, but complaining about macro idiom that is older than I am, and is used to make macros more sanitary is bleh.

Reply:
https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines/avoid-do-while.html

I wanted to mention that clang-tidy has option to disable this check for macros so users encountering this problem can fix it in their own projects if they wish. I can't think of any reason someone would want to keep this warning for 1st party macros but would be upset that Catch2 uses this pattern. As you said it's a very well-established pattern.